Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -35,102 +35,137 @@ namespace Google.Api.Generator.Generation
                             foreach (var resourceName in signature.ResourceNames)
                             {
-                                yield return resourceName.SyncPaginatedMethod;
-                                yield return resourceName.AsyncPaginatedMethod;
+                                yield return resourceName.SyncPaginatedSnippet;
+                                yield return resourceName.AsyncPaginatedSnippet;
                             }
                         }
                         break;
                     case MethodDetails.ServerStreaming _:
-                        yield return methodDef.ServerStreamingRequestMethod;
+                        yield return methodDef.ServerStreamingRequestSnippet;
                         foreach (var signature in methodDef.Signatures)
                         {
-                            yield return signature.ServerStreamingMethod;
+                            yield return signature.ServerStreamingSnippet;
                             foreach (var resourceName in signature.ResourceNames)
                             {
-                                yield return resourceName.ServerStreamingMethod;
+                                yield return resourceName.ServerStreamingSnippet;
                             }
                         }
                         break;
                     case MethodDetails.BidiStreaming _:
-                        yield return methodDef.BidiStreamingMethod;
+                        yield return methodDef.BidiStreamingSnippet;
                         break;
                 }
             }
         }
 
+        private SnippetCodeGenerator(SnippetDef snippetDef) => 
+            Snippet = snippetDef;
+
+        private SnippetDef Snippet { get; }
+        public string SnippetName => Snippet.SnippetName;
+
+        public CompilationUnitSyntax Generate(SourceFileContext ctx) =>
+            ctx.CreateCompilationUnit(Snippet.StandaloneSnippet(ctx));
+
+        private class SnippetDef
+        {
+            public string SnippetName { get; }
+            private ServiceDetails Svc { get; }
+            private Func<SourceFileContext, MethodDeclarationSyntax> MethodGenerator { get; }
+
+            public SnippetDef(string snippetName, ServiceDetails svc, Func<SourceFileContext, MethodDeclarationSyntax> methodGenerator) =>
+                (SnippetName, Svc, MethodGenerator) = (snippetName, svc, methodGenerator);
+
+            public MethodDeclarationSyntax GenerateMethod(SourceFileContext ctx) => MethodGenerator(ctx);
+
+            public NamespaceDeclarationSyntax StandaloneSnippet(SourceFileContext ctx)
+            {
+                var ns = Namespace(Svc.SnippetsNamespace);
+                using (ctx.InNamespace(ns))
+                {
+                    var cls = Class(Public | Sealed | Partial, Svc.StandaloneSnippetsTyp)
+                        .WithXmlDoc(XmlDoc.Summary("Generated snippet."));
```

Comment:
As noted in the design doc, I suspect we don't want a summary per partial class.